### PR TITLE
[pulsar-admin] add read-timeout to blocked admin-request

### LINKED ISSUE #4447: Broker gets stuck when getting partitioned stats
**Describe the bug**

Brokers in our cluster get stuck when executing `internalGetPartitionedStats`:
https://github.com/apache/pulsar/blob/3cfa81250e49b24b4b195f8e2d510fcd24b93745/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java#L609-L610

thread dump:
https://gist.github.com/nkurihar/5d90ce683802390cdec2d9fb65cc0297
```
"pulsar-web-28-32" #531 prio=5 os_prio=0 tid=0x00007f31a4001000 nid=0x3b69 waiting on condition [0x00007f30db1c6000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00007f512c8e2d80> (a java.util.concurrent.CompletableFuture$Signaller)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.CompletableFuture$Signaller.block(CompletableFuture.java:1693)
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3323)
	at java.util.concurrent.CompletableFuture.waitingGet(CompletableFuture.java:1729)
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1895)
	at org.apache.pulsar.client.admin.internal.TopicsImpl.getStats(TopicsImpl.java:380)
	at org.apache.pulsar.broker.admin.impl.PersistentTopicsBase.internalGetPartitionedStats(PersistentTopicsBase.java:610)
	at org.apache.pulsar.broker.admin.v1.PersistentTopics.getPartitionedStats(PersistentTopics.java:279)
	at sun.reflect.GeneratedMethodAccessor56.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:845)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1689)
	at org.apache.pulsar.broker.web.ResponseHandlerFilter.doFilter(ResponseHandlerFilter.java:53)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1676)
	at org.apache.pulsar.broker.web.AuthenticationFilter.doFilter(AuthenticationFilter.java:74)
	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1676)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:581)
	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:224)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1180)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:511)
	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:185)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1112)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:213)
	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:119)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:134)
	at org.eclipse.jetty.server.Server.handle(Server.java:524)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:319)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:253)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
	at org.eclipse.jetty.io.ssl.SslConnection.onFillable(SslConnection.java:202)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:273)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:95)
	at org.eclipse.jetty.io.SelectChannelEndPoint$2.run(SelectChannelEndPoint.java:93)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.executeProduceConsume(ExecuteProduceConsume.java:303)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.produceConsume(ExecuteProduceConsume.java:148)
	at org.eclipse.jetty.util.thread.strategy.ExecuteProduceConsume.run(ExecuteProduceConsume.java:136)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

**To Reproduce**
We haven't found the way to reproduce yet.

**Additional context**
Broker OS: CentOS Linux release 7.6.1810
Broker version: 2.2.1
Broker spec: Real server / 2.10GHz / 2CPU / 256GBMEM / SATA SSD 240GB x1 / 10G Base-T*2port

### PR DESCRIPTION
**Pull request by rdhabalia**
### Motivation
It fixes #4447 and unblocks broker's admin thread and avoids possible deadlock.

### CODE REVIEW

**Comment by rdhabalia** [comment]
rerun java8 tests

File: pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java [hunk 1]
```diff
@@ -187,22 +187,23 @@ public PulsarAdmin(String serviceUrl,
...
-        this.nonPersistentTopics = new NonPersistentTopicsImpl(root, auth);
-        this.resourceQuotas = new ResourceQuotasImpl(root, auth);
-        this.lookups = new LookupImpl(root, auth, useTls);
-        this.functions = new FunctionsImpl(root, auth, httpAsyncClient);
-        this.sources = new SourcesImpl(root, auth, httpAsyncClient);
-        this.sinks = new SinksImpl(root, auth, httpAsyncClient);
-        this.worker = new WorkerImpl(root, auth);
-        this.schemas = new SchemasImpl(root, auth);
-        this.bookies = new BookiesImpl(root, auth);
+        long readTimeoutMs = readTimeoutUnit.toMillis(this.readTimeout);
```
Comment (nkurihar) [inline pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: I haven't found the line which sets `readTimeout` in https://github.com/apache/pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java , so `readTimeout` seems to be set to the default value (60 seconds) for Brokers.
https://github.com/apache/pulsar/blob/b5d779db5f674469797873f2c5627d4805aae082/pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java#L72

Can you make it configurable in `broker.conf`?
(I feel 60 seconds is too large for Brokers)

Comment (rdhabalia) [reply pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/PulsarAdmin.java]: sure, I will fix it.

**Comment by sijie** [comment]
ping @rdhabalia

**Comment by rdhabalia** [comment]
@nkurihar most of the admin-sync call makes blocking zk call so, configured admin-readtimeout as `zkOperationTimeout`. Please let me know if we have different thought on it.

 @sijie 
> I see some places are using connectTimeoutMs and some places are using readTimeoutMs. Can we be consistent across all the places?

yes, I have fixed it.

**Comment by rdhabalia** [comment]
rerun java8 tests

**Comment by nkurihar** [comment]
@rdhabalia 
Thank you for making `readTimeout` configurable!

minor comment:
How about adding comments to `zooKeeperOperationTimeoutSeconds` in `broker.conf` to explain that this value is also used as admin-readtimeout in Brokers?
https://github.com/apache/pulsar/blob/a19522130c4e6c848b6d80ae28a0d016fab8210c/conf/broker.conf#L64-L65

**Comment by nkurihar** [comment]
rerun java8 tests

**Comment by codelipenghui** [comment]
@rdhabalia do we need this issue for 2.4.0? or can we move it to 2.5.0?

**Comment by rdhabalia** [comment]
yes, it's ready to merge.

**Comment by codelipenghui** [comment]
@sijie Please help review again.

**Comment by sijie** [comment]
@nkurihar can you review and merge it?

**Pull request merged**